src/pages/docs/guides/chat/bring-own-database.mdx

```diff
@@ -0,0 +1,123 @@
+
+Ably can forward messages from chat room channels to an Ably Queue, which you can then consume from your own servers to save messages to your own database. Read the guide on [Ably queues](/docs/platform/integrations/queues) for more details on how to setup the queue integration with Ably.
+
+You can use AMQP or STOMP to consume messages from an Ably queue. Ably ensures that each message is delivered to only one consumer even if multiple consumers are connected.
+
+Benefits of using an Ably queue:
+
+- You can consume it from your servers, meaning overall this is fault-tolerant. Ably takes care of the complexity of maintaining a queue.
+- You can use multiple queues and configure which channels go to which queue (use `.*::\$chat$` regex to match all chat rooms).
+- If your systems suffer any downtime, you will not miss anything (up to the queue max size).
+
+You need to consider:
+- During peak times you may need to scale up your consumers to avoid overloading the queue past the maximum queue length allowed.
+- Each message has a TTL in the queue.
+- Oldest messages are dropped if the maximum queue length is exceeded.
+
+## Publishing via your own servers
+
+Change the publish path: instead of publishing Chat messages, updates, and deletes to Ably directly, proxy them through your own server. This gives you the opportunity to also save the messages as they are produced, and also apply different validation schemes if needed.
+
+Benefits:
+- Full control over publishing.
+- Opportunity to add extra validation before publishing to Ably.
+- You can publish messages directly via the Chat REST API, and avoid having to encode/decode Chat Messages to and from Ably Pub/Sub messages. You can bypass using an SDK entirely or you can use the Chat SDK for publishing.
+
+You need to consider:
+- You need to handle updates and deletes on your own.
+- Storing message reactions can be difficult since you will not have access to the aggregate (summaries) Ably provides.
+- Your own servers are in the middle of the message publish path, so they can become a bottleneck in availability and will add latency in the publish path.
+- Your own servers will need to handle the scale you operate at for realtime publishes.
+
+## A note about the Chat History endpoint <a id="chat-history-api"/>
+
+The chat room history endpoint is a paginated HTTP endpoint that allows you to retrieve messages from a chat room. Its intended use is to retrieve messages for pre-filling a chat window, and not for ingesting messages into other systems.
+
+The history API returns messages in their canonical global order. For each message, only the latest version of the message is returned. The history endpoint is not a changelog, but a snapshot of the messages in the room at the time the request is made.
+
+It is not suitable for ingesting messages into other systems, because polling it regularly will not converge to the correct state. It will miss updates and deletes of older messages.
```

Comment:
True, but I think it has its uses. If you're doing 1:1 support chats, you could, in theory, query history once the chat is done to ingest it in one go.